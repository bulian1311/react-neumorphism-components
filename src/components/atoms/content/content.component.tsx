import React, { FC } from 'react';
import { StyledContent } from './content.styled';
import { ContentProps } from './content.types';

export const Content: FC<ContentProps> = ({ children, size }) => {
  return <StyledContent>{children}</StyledContent>;
};

Content.displayName = 'Content';

Content.defaultProps = {};

Content.propTypes = {};
