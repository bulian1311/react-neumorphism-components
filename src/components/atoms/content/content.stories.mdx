import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as stories from './content.stories.tsx';
import { Content } from './content.component.tsx';

<Meta title="Atoms/Content" component={Content} />

# Content

This component handles WYSIWYG/plain HTML content. The example below demonstrates the formatting of different HTML elements.

<ArgsTable of={Content} />

## Example

<Canvas>
  <Story story={stories.Content} name="Content" />
</Canvas>
