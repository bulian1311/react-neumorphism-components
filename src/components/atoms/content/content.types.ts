export type ContentProps = {
  size?: 'small' | 'medium' | 'large';
};
