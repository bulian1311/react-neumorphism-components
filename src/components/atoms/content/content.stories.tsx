import React from 'react';
import { Story } from '@storybook/react';

import { Content as ContentComponent } from './content.component';

export const Template: Story = () => (
  <ContentComponent>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Recusandae magnam
      adipisci incidunt expedita!
    </p>
  </ContentComponent>
);

export const Content = Template.bind({});

Content.args = {};

Content.argTypes = {};
