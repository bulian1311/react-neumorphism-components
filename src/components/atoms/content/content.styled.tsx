import styled, { css } from 'styled-components';
import { ContentProps } from './content.types';

const dinamicStyles = (props: ContentProps) => {
  let fontSize: number;
  switch (props.size) {
    case 'small':
      fontSize = 0.8;
      break;
    case 'medium':
      fontSize = 1;
      break;
    case 'large':
      fontSize = 1.2;
      break;
    default:
      fontSize = 1;
  }

  return css`
    font-size: ${fontSize}rem;
  `;
};

export const StyledContent = styled.div<ContentProps>`
  ${dinamicStyles}
`;
