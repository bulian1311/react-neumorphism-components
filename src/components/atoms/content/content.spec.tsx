import React from 'react';
import { Content } from './content.component';
import { render } from '@testing-library/react';

describe('Box', () => {
  it('matches snapshot', () => {
    const { container } = render(<Content>Content</Content>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
