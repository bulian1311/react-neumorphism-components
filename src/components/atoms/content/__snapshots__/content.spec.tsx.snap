// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box matches snapshot 1`] = `
<div
  class="sc-bdvvtL ghgWOr"
>
  Content
</div>
`;
